src/app/examples/todos/todos.reducer.ts

```diff
@@ -74,8 +74,9 @@ export function todosReducer(
             id: uuid(),
```

Comment:
The `uuid()` call in reducer is the main offender. Reducer should never behave in in-deterministic fashion. The solution for this would be to change corresponding action to generate id during the action creation.

Reply:
May better solution is to move the uuid call to "onAddTodo" method of todos.component?  

```
this.store.dispatch(new ActionTodosAdd({id: uuid(), name: this.newTodo, done: false } as Todo));
```

Reply:
@timdeschryver what would you suggest? I guess making ID creation part of the action itself would make in this case?

Reply:
Yep! In your components you don't really care about the `id` (and you would also have to import the `uuid` function where you need it), so I would suggest to generate the `id` inside the action creator.